bin/protoc-gen-ruby_gapic

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env ruby
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+$LOAD_PATH << ::File.expand_path('../../lib', __FILE__)
+require 'google/gapic/schema'
+require 'google/gapic/generator'
+require 'protobuf/descriptors'
+
+# Ensure that no encoding conversions are done on STDIN and STDOUT since
+# we are passing binary data back and forth. Otherwise these streams
+# will be mangled on Windows.
+STDIN.binmode
+STDOUT.binmode
+
+# Get the CodeGeneratorRequest.
+request_bytes = STDIN.read
+request = Google::Protobuf::Compiler::CodeGeneratorRequest.decode(request_bytes)
+
+# Determine the appropriate package.
+# This generator uses a slightly different mechanism for determining
+# which files to generate; it tracks at package level rather than file
+# level.
+files_to_generate = request.proto_file.select do |file|
+  request.file_to_generate.include? file.name
+end
+package = files_to_generate
+  .reduce(files_to_generate[0].package.split('.')) do |answer, file|
+    parts = file.package.split('.')
+    parts.each_with_index do |part, i|
+      return parts[0..i] if answer[i] != part
```

Comment:
I suspect this `return` will raise a LocalJumpError since it's not in an actual method (and none of the blocks here are lambdas). Which level do you want to break out of?

Reply:
Ah this is unneeded since we are simply using the CodeGeneratorRequest.file_to_generate to determine which files need to be generated rather than using package.